resources/osl_synapse.rb

```diff
@@ -0,0 +1,95 @@
+resource_name :osl_synapse
+provides :osl_synapse
+unified_mode true
+
+default_action :create
+
+property :app_services, Array, default: []
+property :config, String
+property :domain, String, name_property: true
+property :name, String
+property :network, String
+property :path, String
+property :pg_host, String
+property :pg_name, String
+property :pg_username, String
+property :pg_password, String
+property :port, Integer, default: 8008
+property :tag, String, default: 'latest'
+property :use_sqlite, [true, false], default: false
+
+action :create do
+  # Check to see if a name was given
+  new_resource.name = "matrix-synapse-#{new_resource.domain}" unless new_resource.name
+  # Get the path to use for this resource
+  new_resource.path = "/opt/synapse-#{new_resource.name}" unless new_resource.path
+  # Set the docker network
+  new_resource.network = "synapse-network-#{new_resource.name}" unless new_resource.network
```

Comment:
These should all be defaults above using `lazy` to set them based on the `domain` property.